core/src/test/java/io/grpc/internal/MessageDeframerTest.java

```diff
@@ -139,7 +196,17 @@ public void payloadSplitBetweenBuffers() {
         Bytes.asList(new byte[] {3, 14, 1, 5, 9, 2, 6}), bytes(producer.getValue().next()));
     verify(listener, atLeastOnce()).bytesRead(anyInt());
     verifyNoMoreInteractions(listener);
-    checkStats(1, 7, 7);
+
+    if (useGzipInflatingBuffer) {
+      checkStats(
```

Comment:
@zhangkun83, can we remove these checks from all but a couple of tests? This test for instance really has nothing to do with stats?

Reply:
It can be argued either way. It's not unreasonable to say that for each case you want to test that the deframer produces the correct output, you would also want to test that it records the correct measurements.

If hard-coding GZIP-derived numbers is the issue, can we try running the input through GZIP and get the actual wire size on the fly?

Reply:
Changed the stats reporting for compressed-streams to exclude gzip metadata (per offline discussion with @zhangkun83). Only the `payloadSplitBetweenBuffers` test still reports stats differently with full-stream compression turned on, as the deflate block adds 2 bytes of overhead for compressing the second buffer.